src/common/test/TestImageUdderTest.cpp

```diff
@@ -0,0 +1,48 @@
+/** Copyright (C) 2016, 2017 European Spallation Source ERIC */
+
+#include <common/TestImageUdder.h>
+#include <test/TestBase.h>
+
+class TestImageUdderTest : public TestBase {
+protected:
+  Udder udder;
+  ESSGeometry geom{udder.ImageWidth, udder.ImageHeight, 1, 1};
+};
+
+TEST_F(TestImageUdderTest, Constructor) {
+  ASSERT_EQ(udder.ImageWidth, 184); // the udder image dimension
+  ASSERT_EQ(udder.ImageHeight, 224); // the udder image dimension
+  ASSERT_FALSE(udder.isCached());
+  ASSERT_EQ(udder.getNumberOfCachedPixels(), 0);
+}
+
+TEST_F(TestImageUdderTest, FirstFewPixels) {
+  auto pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 123);
+
+  pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 124);
+}
+
+TEST_F(TestImageUdderTest, CachePixels) {
+  ASSERT_FALSE(udder.isCached());
+  udder.cachePixels(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_TRUE(udder.isCached());
+  auto pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 123);
+
+  pixel = udder.getPixel(udder.ImageWidth, udder.ImageHeight, &geom);
+  ASSERT_EQ(pixel, 124);
+}
+
+TEST_F(TestImageUdderTest, CachePixelsWrap) {
+  uint32_t FirstPixel, LastPixel;
```

Comment:
Variable name is not consistent, as `udder` and `geom` are lowercase, while `FirstPixel` and `LastPixel` are camel case.

Reply:
Ok, not really sure I care about this in the unit tests, but since it is you, I'll fix this ;-)